pkg/providers/nutanix/validator.go

```diff
@@ -59,6 +60,26 @@ func (v *Validator) ValidateDatacenterConfig(ctx context.Context, config *anywhe
 		return err
 	}
 
+	if err := v.validateCredentialRef(config); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (v *Validator) validateCredentialRef(config *anywherev1.NutanixDatacenterConfig) error {
+	if config.Spec.CredentialRef == nil {
+		return fmt.Errorf("credentialRef must be provided")
+	}
```

Comment:
How does this affect cluster upgrades where the old datacenter configs didn't have the creds ref?

Reply:
they will have to be reconciled.